### PR TITLE
Pauline

### LINKED ISSUE #443: Create the News page
#  Create the News page Hero with news tabs
---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user I want a tabbed news section - use ngSwitch to implements the tabs.

### Description

Create a dummy list of articles with the filter properties provided in the design. When a user clicks on a tab it should filter the news list below.

Create the tabs using ngSwitch

for the news articles reuse #411 

Screenshot 2023-03-09 at 09 33 45

Please take note of all the design features

#### Link to full design:
```
https://xd.adobe.com/view/7b72572b-b40a-4618-bfa6-5500ab28f13b-2e21/screen/f57793f4-a53c-4628-9f7a-9c2f6c0fd8ec
```

### Technical Analysis

This component will be created in  libs/pages/elewa/news

The component will be called elewa-news-section

### Acceptance Criteria

1. When rendered the component is pixel perfect compared to the design above.
2. The component is clean and does not have more than 50 lines
3. The component SCSS uses CSS variables for colour, font, ... Connected to the main style.scss
4. The code is isolated to the libs/page/home package, unless a justified reason is specified.
5. The component is aligned using CSS flex
6. The design is responsive for screen, tablet and phone
7. The image renders fast and has multiple srcsets

### PR DESCRIPTION
**Pull request by pauleen123**
# Description

This pull request is part of the work to create the news section page where a user can view and filter different articles.

To achieve this, we needed to:

- Re-use issue #411 
- Create a filter function that filters through the tabs for specific news
- 
Fixes #443 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Screenshot (optional)
large screen view for the all news tab
largescreenview

large screen view on the elewa tab
largeviewonelewatab

mobile view on the all news tab
mobileview1

mobile view on the i-talanta tab
mobileview2

# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by pauleen123** [comment]
hey @IanOdhiambo9 kindly review my pull reques on this issue

File: libs/features/components/ui-lists/src/lib/elewa-group-article-list/elewa-group-article-list.component.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { Component, Input } from '@angular/core';
-export interface Article { image: string; timestamp: string; title: string; body: string;} //import this interface in the component you will be reusing article-list in. e.g import { Article } from 'libs/features/components/ui-lists/src/lib/elewa-group-article-list/elewa-group-article-list.component';
+export interface Article { image: string; timestamp: string; title: string; body: string; topic: string;} //import this interface in the component you will be reusing article-list in. e.g import { Article } from 'libs/features/components/ui-lists/src/lib/elewa-group-article-list/elewa-group-article-list.component';
```
Comment (IanOdhiambo9) [inline libs/features/components/ui-lists/src/lib/elewa-group-article-list/elewa-group-article-list.component.ts:2]: Move interface to an interface file inside this lib. models/article.interface.ts

File: libs/pages/elewa/news/src/lib/components/elewa-news-section/elewa-news-section.component.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+<div class="container">
+    <h2>News & Stories</h2>
+    <nav class="tabs">
+      <a class="tab" (click)="selectedTab='all-news'; filterArticles();" [class.active]="selectedTab==='all-news'">All News</a>
+      <a class="tab" (click)="selectedTab='elewa'; filterArticles();" [class.active]="selectedTab==='elewa'">Elewa</a>
+      <a class="tab" (click)="selectedTab='italanta'; filterArticles();" [class.active]="selectedTab==='italanta'">Italanta</a>
+      <a class="tab" (click)="selectedTab='press'; filterArticles();" [class.active]="selectedTab==='press'">Press</a>
+      <a class="tab" (click)="selectedTab='venture-labs'; filterArticles();" [class.active]="selectedTab==='venture-labs'">Venture Labs</a>
+    </nav>
```
Comment (IanOdhiambo9) [inline libs/pages/elewa/news/src/lib/components/elewa-news-section/elewa-news-section.component.html:11]: No need for nav, use a loop here and make this a button

**Review by IanOdhiambo9 — CHANGES_REQUESTED** [review]

**Comment by IanOdhiambo9** [comment]
Why is this identical to https://github.com/italanta/elewa-group/pull/466 ??

**Comment by pauleen123** [comment]
hey @IanOdhiambo9 kindly review my updated PR as per the requested changes

**Pull request closed without merging**